packages/core-data/src/resolvers.js

```diff
@@ -543,13 +543,12 @@ export const __experimentalGetCurrentGlobalStylesId =
 		const globalStylesURL =
 			activeThemes?.[ 0 ]?._links?.[ 'wp:user-global-styles' ]?.[ 0 ]
 				?.href;
-		if ( globalStylesURL ) {
-			const globalStylesObject = await apiFetch( {
-				url: globalStylesURL,
-			} );
-			dispatch.__experimentalReceiveCurrentGlobalStylesId(
-				globalStylesObject.id
-			);
+		if ( ! globalStylesURL ) {
+			return;
+		}
+		const id = +globalStylesURL.split( '/' ).pop();
```

Comment:
I'm a little nervous about the `split` and `pop` as it assumes the href from the link will have the same format (for example, it'll not have any query strings). That could be something that changes in the future, and If it ever had a query string the `id` would end up as `NaN`. Maybe a fairly loose regex would be safer? I guess it needs to take into account that the URL might have some differences on different WordPress installations.

Other than that, the PR looks good to me, the added preload paths bring the post editor more in line with the site editor, which would be a good thing to do with the editor unification work.

Reply:
I guess we should also follow up with getting https://github.com/WordPress/gutenberg/pull/61095 merged, so replicating whatever changes are made to this code in the other PR.

Reply:
Thanks 👍 

I've updated this to use a very basic regex to extract the id in https://github.com/WordPress/gutenberg/pull/63071/commits/d96ff045eea11f5577284fe4fe07d9d1827f09c1.

The regex isn't 100% future-proof either, however, so long as the ID is the last segment of the URL before a possible query string, it should be good enough. Are you happy with that?